docs/platform/web-project.md

```diff
@@ -30,72 +30,143 @@ a _feature_ grouping it together with its dependencies,
+folder for bundle implementation [modules](https://maven.apache.org/guides/mini/guide-multiple-modules.html) and an `assemblies` folder for modules concerning the provisioning of those bundles.
 
-```shell
-mvn archetype:generate ...
+If the project contemplates multiple bundles then `impl` will itself include multiple 
+modules, one for each bundle. If the project only has a single bundle then it is OK for `impl` to 
+directly contain the bundle implementation module without the need for a sub module.
+
+A provisioning module, in the context of a Pentaho OSGI project, is a module that automates, 
+as much as possible, the creation of a feature file and potentially the KAR file that allows that feature 
+to be easily deployed. 
+One reason why multiple provisioning modules might exist is that it may desired to deploy the bundle(s) in 
+different applications such as pentaho-server and pdi-client (spoon). The different 
+targets may warrant different configurations or dependency provisioning and thus different features 
+and KAR files.
+
+In order to take advantadge of Pentaho conventions and build pipeline, the root project should have
+the [`pentaho-ce-bundle-parent-pom`](https://github.com/pentaho/maven-parent-poms/blob/7.1.0.0-R/pentaho-ce-parent-pom/pentaho-ce-jar-parent-pom/pentaho-ce-bundle-parent-pom/pom.xml) as its parent project.
+
+
+### Bundle modules
+
+Consider the following example folder structure for `impl`:
+
+```
+|____rootProject
+| |____impl
+| | |____src
+| | | |____main
+| | | | |____config
+| | | | | |____javascript
+| | | | | | |____osgi
+| | | | |____javascript
+| | | | | |____web
+| | | | |____resources
+| | | | | |____META-INF
+| | | | | | |____js
+| | | | | |____OSGI-INF
+| | | | | | |____blueprint
 ```
 
->  - Maven project structure starting with the maven archetype. 
->    - _Do not inherit from pentaho parent POMs_
->    - _We need to check where/how to publish the archetype_
+On this example case we have a single bundle and as such its source folder is directly under the `impl` 
+folder  (`impl/src`) instead of being on a dedicated sub module (`impl/myBundleModule/src`).
 
-## Explain the folder structure
+The `impl/src/main/config/javascript/osgi` folder contains the configuration file for the optimisation that
+is to be performed by the [RequireJS optimizer](http://requirejs.org/docs/optimization.html) when building 
+the bundle. If the configuration file exists but does not specify any configuration option then by default 
+each javascript file is minimised in place and source maps are generated.
 
->  - Explain the folder structure
->    - What are the assemblies (generating kar) 
->    - What are the impl (the actual viz)
->      - ~Blueprint~ Keep it to a minimum. Just mentioned that this file is used to specify where (URL) the JS code will be available from
+The `impl/src/main/resources/META-INF/js` folder is where the [`package.json`](./web-package) descriptor should 
```

Comment:
The `impl/src/main/resources/META-INF/js` folder is where the [`package.json`](./web-package) descriptor should be placed. 
(Remove the preposition)